Lib/test/test_long.py

```diff
@@ -956,6 +956,16 @@ def test_huge_rshift_of_huge(self, size):
         self.assertEqual(huge >> (sys.maxsize + 1), (1 << 499) + 5)
         self.assertEqual(huge >> (sys.maxsize + 1000), 0)
 
+    @support.cpython_only
+    def test_small_ints_in_huge_calculation(self):
+        a = 2 ** SHIFT
+        b = 2 ** SHIFT - 1
+        c = 2 ** SHIFT + 1
+        d = -a
+        self.assertIsNot(a + b, 2 ** (SHIFT + 1) - 1)
```

Comment:
I don't see the point of this test. I prefer to have assertIs() tests to ensure that optimizations work.